docs/proposals/versioning.md

```diff
@@ -0,0 +1,25 @@
+## Plan for versioning APBs
+
+### Current status:
+* Currently all APBs are 0.1.0
+  * This is stored in a label on the image titled “com.redhat.apb.version”
+* Plan is to bump all APBs to a newer version to coincide with 3.7 Broker
+
+### Questions:
+* Should this be a major version bump? I.e. 1.0.0, or minor would be 0.2.0
+* Do we tag the images with the version of the broker or APB spec version? Should they match?
+
+### Thoughts:
+* Makes sense to do a major version bump because the version has not changed since we were ansibleapp.
+* Since the ‘plan’ format of the APB is not likely to change anytime soon it would make sense to establish the schema of the spec in a major bump. There are still some new APB developers who find old examples and try to use it without plans.
+* 1.x.x APBs work on ASB/OC 3.7
+* 0.x.x APBs are < ASB 3.6 
+* Image tags should match whatever version number we choose. This would be a pro for versioning APBs in the same vein as ASB. i.e. 3.7 ASB can launch images tagged with 3.7.
```

Comment:
So there is definitely value here with keeping versions locked in with OCP it makes it very easy to see what works with what. But this also means that we can NOT make any major changes in a given release. Let's say at the beginning of 3.8 development we say that APB will now be versioned 3.8. If we do a change the equivalent to plans mid development cycle that would require us to go to 4.0 which means APB 4.0 would work with ASB/OCP 3.8 which is quite confusing.

If the versions were NOT locked down to OCP, then we could independently bump ASB/APB versions to ensure that we know APB x.y is compatible with ASB x.

Reply:
+1. I think you make good points for not locking the version to OCP. I'm in agreement as I expect the APB spec to develop major feature changes before OCP does.